File: commands_test.go
package main

import "testing"

func TestParseCommands(t *testing.T) {
	tests := []struct {
		input, name, since, text string
		ok                       bool
	}{
		{"/start abc", "start", "abc", "", true}, {" /PAUSE ", "pause", "", "", true},
		{"keep it short", "steer", "", "keep it short", true}, {"/wat", "", "", "", false}, {" ", "", "", "", false},
	}
	for _, test := range tests {
		got, ok := ParseCommand(test.input)
		if ok != test.ok || got.Name != test.name || got.Since != test.since || got.Text != test.text {
			t.Errorf("ParseCommand(%q) = %#v, %v", test.input, got, ok)
		}
	}
}

File: test/commands.test.ts
import { describe, expect, it } from "vitest";
import { parseCommand } from "../src/commands.js";

describe("parseCommand", () => {
  it("parses lifecycle commands", () => {
    expect(parseCommand("/start abc")).toEqual({ name: "start", since: "abc" });
    expect(parseCommand(" /PAUSE ")).toEqual({ name: "pause" });
    expect(parseCommand("/stop")).toEqual({ name: "stop" });
  });

  it("treats plain text as steering", () => {
    expect(parseCommand("keep it very short")).toEqual({ name: "steer", text: "keep it very short" });
  });

  it("rejects empty and unknown commands", () => {
    expect(parseCommand(" ")).toBeUndefined();
    expect(parseCommand("/wat")).toBeUndefined();
  });
});
